src/views/scans/showAggregateReportModal.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  Modal,
+  ModalVariant,
+  Title
+} from '@patternfly/react-core';
+import { helpers } from '../../helpers';
+import { type ReportAggregateDiagnosticsType, type ReportAggregateResultsType } from '../../types/types';
+
+type PartialReportsAggregateResponse = {
+  results?: Partial<ReportAggregateResultsType>;
+  diagnostics?: Partial<ReportAggregateDiagnosticsType>;
+};
+
+type Report = {
+  id: number;
+  report: PartialReportsAggregateResponse;
+};
+
+interface ShowAggregateReportModalProps {
+  isOpen: boolean;
+  report?: Report;
+  onClose?: () => void;
+  actions?: React.ReactNode[];
+}
+
+/**
+ * A filter list for displaying specific report properties. Order does not represent display sort.
+ */
+const statsFilter = [
+  'ansible_hosts_all',
+  'instances_hypervisor',
+  'instances_physical',
+  'instances_virtual',
+  'openshift_cores',
+  'socket_pairs',
+  'system_creation_date_average',
+  'vmware_hosts',
+  'inspect_result_status_failed',
+  'inspect_result_status_success',
+  'inspect_result_status_unknown',
+  'inspect_result_status_unreachable',
+  'missing_pem_files',
+  'missing_system_creation_date',
+  'missing_system_purpose'
+];
+
+/**
+ * Apply list format, sort, and filter to report
+ */
+const formatSortFilterReportStats = (
+  report: PartialReportsAggregateResponse = {},
+  { filter = statsFilter }: { filter?: string[] } = {}
+) => {
+  const { results: resultStats = {}, diagnostics: diagnosticStats = {} } = report;
+
+  return [...(resultStats && Object.entries(resultStats)), ...(diagnosticStats && Object.entries(diagnosticStats))]
```

Comment:
Not a blocker, but have you considered keeping `results` and `diagnostics` separated in the UI instead of merging and sorting them together? Just wondering if there would be any downside or extra complexity in doing so.

Reply:
We did...
- Reason... since we ended up using a single loop to do the component display we merged the two lists
- Considered... we considered keeping them separate and running them each through a support function which would support an iteration where dual columns are displayed... something like

```
const formatFilterSort = (arr): [string, unknown][] =>
    arr
      .filter(...
      .sort(...
      .map(...);

  const results = formatFilterSort(RESULT_OBJ);
  const diagnostics = formatFilterSort(DIAGNOSTIC_OBJ);

  return {
    hasResults: results.length > 0,
    hasDiagnostics: diagnostics.length > 0,
    results,
    diagnostics
  };
```